pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1554 @@
+				ok = false
+			}
+		}
+	}
+
+	return ok, nil
+}
+
+func (k *Operation) FromGnostic(g *openapi_v2.Operation) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+
+	ok = true
+
+	if nok, err := k.VendorExtensible.FromGnostic(g.VendorExtension); err != nil {
+		return false, err
+	} else if !nok {
+		ok = false
+	}
+
+	if nok, err := k.OperationProps.FromGnostic(g); err != nil {
+		return false, err
+	} else if !nok {
+		ok = false
+	}
+
+	return ok, nil
+}
+
+func (k *OperationProps) FromGnostic(g *openapi_v2.Operation) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+
+	ok = true
+
+	k.Description = g.Description
+	k.Consumes = g.Consumes
+	k.Produces = g.Produces
+	k.Schemes = g.Schemes
+	k.Tags = g.Tags
+	k.Summary = g.Summary
+
+	if g.ExternalDocs != nil {
+		k.ExternalDocs = &ExternalDocumentation{}
+		if nok, err := k.ExternalDocs.FromGnostic(g.ExternalDocs); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+	}
+
+	k.ID = g.OperationId
+	k.Deprecated = g.Deprecated
+
+	if g.Security != nil {
+		k.Security = make([]map[string][]string, len(g.Security))
+		for i, v := range g.Security {
+			if v.AdditionalProperties == nil {
+				continue
+			}
+
+			k.Security[i] = make(map[string][]string, len(v.AdditionalProperties))
+			converted := k.Security[i]
+			for _, p := range v.AdditionalProperties {
+				if p.Value != nil {
+					converted[p.Name] = p.Value.Value
+				} else {
+					converted[p.Name] = nil
+				}
+			}
+		}
+	}
+
+	if g.Parameters != nil {
+		k.Parameters = make([]Parameter, len(g.Parameters))
+		for i, v := range g.Parameters {
+			if v == nil {
+				continue
+			} else if nok, err := k.Parameters[i].FromGnosticParamtersItem(v); err != nil {
+				return false, err
+			} else if !nok {
+				ok = false
+			}
+		}
+	}
+
+	if g.Responses != nil {
+		k.Responses = &Responses{}
+		if nok, err := k.Responses.FromGnostic(g.Responses); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+	}
+
+	return ok, nil
+}
+
+// Responses
+
+func (k *Responses) FromGnostic(g *openapi_v2.Responses) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	}
+	ok = true
+	if nok, err := k.VendorExtensible.FromGnostic(g.VendorExtension); err != nil {
+		return false, err
+	} else if !nok {
+		ok = false
+	}
+
+	if nok, err := k.ResponsesProps.FromGnostic(g); err != nil {
+		return false, err
+	} else if !nok {
+		ok = false
+	}
+
+	return ok, nil
+}
+
+func (k *ResponsesProps) FromGnostic(g *openapi_v2.Responses) (ok bool, err error) {
+	if g == nil {
+		return true, nil
+	} else if g.ResponseCode == nil {
+		return ok, nil
+	}
+
+	ok = true
+	for _, v := range g.ResponseCode {
+		if v == nil {
+			continue
+		}
+		if v.Name == "default" {
+			k.Default = &Response{}
+			if nok, err := k.Default.FromGnosticResponseValue(v.Value); err != nil {
+				return false, err
+			} else if !nok {
+				ok = false
+			}
+		} else if nk, err := strconv.Atoi(v.Name); err == nil {
```

Comment:
if err is non-nil, this should either return an error or at least set `ok` to false (and add a testcase to cover this)

Reply:
done